.github/workflows/ci.yml

```diff
@@ -177,23 +204,33 @@ jobs:
 
           echo "config_key=$config_key" >> $GITHUB_OUTPUT
 
-      - name: Push updated challenge scores
-        if: github.event_name == 'push'
+      - id: push_challenge_scores
+        name: Push updated challenge scores
         run: |
           score_file="tests/challenges/current_score.json"
-
+          challenge_scores_branch="challenge-scores-${{ github.event.pull_request.number }}"
+          echo "challenge_scores_branch=$challenge_scores_branch" >> $GITHUB_OUTPUT
+          
           if ! git diff --quiet $score_file; then
             git add $score_file
             git commit -m "Update challenge scores"
-            git push origin HEAD:${{ github.ref_name }}
+            if [ "${{ startsWith(github.event_name, 'pull_request') }}" = "true" ]; then
+              git checkout -b $challenge_scores_branch
+              git remote add base_repo \
+                https://github.com/${{ github.event.pull_request.base.repo.owner.login }}/${{ github.event.pull_request.base.repo.name }}.git
+              git push -f base_repo $challenge_scores_branch
+              echo "updated_score=true" >> $GITHUB_OUTPUT
+            else
+              git push origin HEAD:${{ github.ref }}
+            fi
           else
             echo "The challenge scores didn't change."
           fi
 
       - id: push_cassettes
         name: Push updated cassettes
         # For pull requests, push updated cassettes even when tests fail
-        if: github.event_name == 'push' || success() || failure()
+        if: always() && (startsWith(github.event_name, 'pull_request') || success())
```

Comment:
Same as above, this will run even if the workflow is canceled. Not sure that's desirable.

Reply:
same as above I suggest keeping it unless you find a better way to write this.